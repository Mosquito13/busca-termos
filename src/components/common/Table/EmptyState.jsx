import { useEffect, useState } from 'react';
import classNames from 'classnames';

import { ReactComponent as EmptyStateImage } from '../../../assets/svg/emptystate.svg';
import Heading4 from '../Heading4';

const EmptyState = () => {
  const [showImage, setShowImage] = useState(false);

  useEffect(() => {
    const imageTimeout = setTimeout(() => setShowImage(true), 3000);

    return () => {
      clearTimeout(imageTimeout);
      setShowImage(false);
    };
  }, []);

  const imageWrapperClasses = classNames(
    'table__empty-state-image-wrapper',
    showImage && 'table__empty-state-image-wrapper--open'
  );

  return (
    <div className="table__empty-state">
      <div className="table__empty-state-message">
        <Heading4 value="Nenhum resultado encontrado." />
      </div>
      <div className={imageWrapperClasses}>
        <div className="table__empty-state-image">
          <EmptyStateImage />
        </div>
      </div>
    </div>
  );
};

export default EmptyState;
