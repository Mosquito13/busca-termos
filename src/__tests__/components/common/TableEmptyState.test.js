import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';

import EmptyState from '../../../components/common/Table/EmptyState';

describe(`${EmptyState.name}`, () => {
  describe('tests', () => {
    let wrapper;

    beforeAll(() => {
      jest.useFakeTimers();

      wrapper = mount(<EmptyState />);
    });

    it('should render without fancy image', () => {
      expect(wrapper.find('.table__empty-state')).toHaveLength(1);
      expect(
        wrapper.find('.table__empty-state-image-wrapper--open')
      ).toHaveLength(0);
    });

    it('should render with fancy image', () => {
      act(() => {
        jest.runAllTimers();
      });
      wrapper.update();
      expect(
        wrapper.find('.table__empty-state-image-wrapper--open')
      ).toHaveLength(1);
    });

    afterAll(() => {
      wrapper.unmount();
    });
  });

  describe('snapshots', () => {
    it('should render', () => {
      const tree = renderer.create(<EmptyState />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
