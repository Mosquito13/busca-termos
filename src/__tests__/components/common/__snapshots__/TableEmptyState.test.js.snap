// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState snapshots should render 1`] = `
<div
  className="table__empty-state"
>
  <div
    className="table__empty-state-message"
  >
    <h4
      className="heading-4"
    >
      Nenhum resultado encontrado.
    </h4>
  </div>
  <div
    className="table__empty-state-image-wrapper"
  >
    <div
      className="table__empty-state-image"
    >
      <svg>
        emptystate.svg
      </svg>
    </div>
  </div>
</div>
`;
